packages/interactivity/src/directives.js

```diff
@@ -236,11 +239,22 @@ export default () => {
 			const defaultEntry = context.find(
 				( { suffix } ) => suffix === 'default'
 			);
-
 			// No change should be made if `defaultEntry` does not exist.
 			const contextStack = useMemo( () => {
 				if ( defaultEntry ) {
 					const { namespace, value } = defaultEntry;
+					// Check that the value is a JSON object. Send a console warning if not.
+					if ( SCRIPT_DEBUG && typeof value !== 'object' ) {
```

Comment:
`null` and `[]` would both pass this check, but I think the intention is to allow "plain" objects.

We have this function in a couple places already in iAPI:

https://github.com/WordPress/gutenberg/blob/2b1b80ea98de13ccd28b537cfaadee0cd601647c/packages/interactivity/src/store.ts#L19-L20
https://github.com/WordPress/gutenberg/blob/2b1b80ea98de13ccd28b537cfaadee0cd601647c/packages/interactivity/src/directives.js#L25-L26

One of those is in this file, let's use it.

Reply:
Oh, I completely missed it. Thanks!